### ISSUE TITLE
Make hiking routes more visible

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#2185]
Its about not all `highway=path`s but those included into `route=hiking` relations - usually its longer more popular and signposted routes of higher significance e.g. national, regional.

The idea is to make them visible at lower zoom levels (at the moment `highway=path` starts from z15) - its very important to be able to get an overview of the route. At z14 about ~2km of the trail fits into the viewport so it makes it very inconvenient to plan a hiking day.

Also makes sense to make the routes a bit more bold at z15- compared to regular hiking trails.

One example of such a route is Lycian Way https://www.openstreetmap.org/relation/51855
https://hiking.waymarkedtrails.org is very good to explore other routes.

E.g. Mapy.cz / Windy Maps gives more visibility to hiking routes and its one of the most appealing points of this app to the hikers.

**Comment by pastk** [comment]
The first question is our generator supports pulling information from type=route relations?

We have a following mapping already
`highway|path|hiking;[highway=path][route=hiking],[highway=path][sac_scale=hiking];;name;int_name;113;`
So `[highway=path][route=hiking]` should transform into `highway-path-hiking` OM type.
However https://www.openstreetmap.org/way/94818955 doesn't seem to have a `highway-path-hiking` shown by the qt app (its only `highway-path` and `psurface-unpaved_bad`)? @vng

**Comment by biodranik** [comment]
The biggest problem now is that making paths visible earlier can greatly pollute many cities and towns. That's why the best idea could be to start working on a separate hiking style where these paths will be visible earlier than on other styles.

**Comment by vng** [comment]
Here is a proposal from support EMail:

Hi,
please add support for colour and osmc:symbol on network=iwn/nwn/rwn/lwn/icn/ncn/rcn/lcn

Example:
yellow hiking
https://www.openstreetmap.org/relation/1578955
https://www.freemap.sk/?map=17/48.660125/20.590385&layers=X
https://en.mapy.cz/turisticka?x=20.5903852&y=48.6601253&z=17


green:white education hiking
https://www.openstreetmap.org/relation/8060109
https://www.freemap.sk/?map=18/48.671900/20.553451&layers=X
https://en.mapy.cz/turisticka?x=20.5534512&y=48.6718997&z=18


red bicycle route
https://www.openstreetmap.org/relation/581944
https://www.freemap.sk/?map=17/48.601634/20.382611&layers=X
https://en.mapy.cz/s/jarapuhara


https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Hiking_routes
https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Cycle_routes

**Comment by vng** [comment]
Also, example of route relations from this way: https://www.openstreetmap.org/way/309621304
network=ncn/lwn

**Comment by pastk** [comment]
First step is importing `route=hiking/bicycle` relations into map data.

**Comment by Explorer256** [comment]
May be included in development `route=foot`?
After all, there are really important foot routes. For example, routes drawn with a line on the pavement.

**Comment by marchukov** [comment]
Just want to note that above proposal may not be ideal for all locations. For example in Czechia there is an extensive network of tourist trails maintained by Czech Tourists Association. Any decent hiking map (either electronics or printed one) is expected to display those with a trail on the map matching its ground colour-coding. [0] is an example of how it looks. 

I guess some country-specific customisation could be helpful.

[0] https://en.mapy.cz/s/kaferavave

**Comment by biodranik** [comment]
Some trails have important info encoded as ref= in their relations. Our users suggest to display that information (e.g. for CAI Italian trails).

**Comment by pastk** [comment]
Route relations (e.g. `route=hiking/foot`) are tricky, e.g. some challenges:
- sections of routes can go by public motor roads, its not just hiking trails and foot paths
- a single trail/path segment may belong to several routes

I think it makes sense to split route relations support into two complimentary cases/approaches:
1. Use existence of relations just to help distinguish between "major" and "regular/minor" trails and set different visibility levels.
Besides belonging to a route we can consider a track "major" when e.g. it has a name, a wikipedia link or a good `surface/smoothness/trail_visibility=` or an easy `sac_scale=`.
2. Comprehensive routes support - render as on-demand color-coded overlay lines, route over them, show distances, elevaton profiles, etc.
So routes should behave much like imported user tracks (and, possibly, guides?) and the implementation could be shared (e.g. it seems like Osmand uses exactly this approach).

Case 1. is much easier to start with.

Its possible to overlay e.g. a dotted red route line over all the roads, paths etc. It'll be useful and shouldn't be hard to implement (the renderer supports drawing many lines for multiple-types features, so we can add e.g. a `route=hiking` type to road features and set its rendering priority higher). But it'll be hard to add other features like routing compared to 2. "user tracks"-like approach, so perhaps its better to focus on a more perspective solution from the beginning.

CC @biodranik @vng

**Comment by biodranik** [comment]
Does it make sense to implement hiking/cycling/other activity relations in a completely different way? For example, export these tracks into e.g. gpx files, and download them in a separate map/bookmarks layer. Or create a separate mwm file with them. Or store them in a custom mwm file section.

From the UX perspective, users need to see more details about the track, photos, and reviews/comments, like it's done in other apps (e.g. AllTrails).

What would be the best approach to take in the longer term? What are the pros and cons of the proposed approach that modify our style system?

**Comment by pastk** [comment]
> Does it make sense to implement hiking/cycling/other activity relations in a completely different way? For example, export these tracks into e.g. gpx files, and download them in a separate map/bookmarks layer. Or create a separate mwm file with them. Or store them in a custom mwm file section.
> 

Yes, this is what I'm thinking of. Treat the routes similar to user imported tracks.
I don't know much how user tracks are implemented atm. They're not a part of the styling system for sure.

**Comment by pastk** [comment]
![Screenshot_20230701-211655.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/93151a4e-21d1-4c1b-81ce-77eff4e0ac5f)

![Screenshot_20230701-211246.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/9fde591d-c80e-4275-92ea-75e8aca95614)

![Screenshot_20230701-211725.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/4411608d-af5a-4a33-af29-0d837f9a3c11)

![Screenshot_20230701-211234.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/15c25c28-cf2c-4f96-93ae-b2640ad09fec)

![Screenshot_20230701-212043.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/c5518c68-31fc-4b17-9662-d25268169291)

**Comment by biodranik** [comment]
Are these colorful lines really usable? They look overloaded and confusing on screenshots. What is your real experience? How to make it better/easier for hikers?

**Comment by endim8** [comment]
a small note: 

In some cases, hiking routes may follow the same path for a while: https://www.freemap.sk/#map=18/45.870008/16.836980&layers=X

the approach taken by mapy.cz is likely done to resolve this issue

**Comment by pastk** [comment]
> @pastk Talking about classifier, yes, better to make separate types like route-hiking-(1,2,3,..) rather then making types like highway-path-hiking, highway-secondary-hiking, etc ...

But how to render the resulting combinations?
E.g. we'd have a feature with `highway-path route-hiking`, or `highway-path-mountain_hiking route-hiking`, or `highway-path route-foot`...
Render in several lines somehow?

**Comment by pastk** [comment]
> Are these colorful lines really usable? They look overloaded and confusing on screenshots. What is your real experience? How to make it better/easier for hikers?

Are you talking about Mapy.cz, Osmand or both? :)

Mapy.cz looks good for someone who is looking for a route. Otherwise it draws too much attention. But in that case one can switch from the "outdoors" to their "basic" style, which doesn't have route lines...

**Comment by vng** [comment]
> But how to render the resulting combinations?

I suppose it will depend on the currently selected style. Clear style - route=* is not drawable; Outdoor (Hike, Bike) - draw solid red/blue/green line with bigger z-index where route=*

**Comment by pastk** [comment]
> > But how to render the resulting combinations?
> 
> I suppose it will depend on the currently selected style. Clear style - route=* is not drawable; Outdoor (Hike, Bike) - draw solid red/blue/green line with bigger z-index where route=*

Draw it over the "standard" `highway-path` dashed line?

**Comment by vng** [comment]
Yes, I don't see other options here. You should draw ```highway-path``` anyway.

**Comment by mahdi1234** [comment]
I think one of the tricky parts is when there's multiple marked trails, I personally don't like the way how Osmand deals with it (randomly selecting one)

![susice](https://github.com/organicmaps/organicmaps/assets/1097047/2e5483ad-f334-4b63-ba8c-37b45bc9ae48)

Freemap handles it way better https://www.freemap.sk/#map=19/49.231174/13.520753&layers=X

In reality it's a mix of four tourist trails https://osm.fit.vutbr.cz/fody/files/4594.jpg and cycling one.

**Comment by biodranik** [comment]
Thanks for the example. Honestly, Freemap's rendering is hard to read too. It would be great to find a better solution.

**Comment by mahdi1234** [comment]
I'm not sure it's any better here it's combining colors together https://openstreetmap.cz/#map=19/49.23094/13.52070&layers=dKG

**Comment by antroyd** [comment]
Providing an online layer to display lonvia routing data (hiking/cycling) could be a first step towards implementing the requested function (see app opentopomap); even though this would be against the general offline character of OrganicMaps. https://f-droid.org/de/packages/org.nitri.opentopo/

**Comment by wzbartczak** [comment]
Displaying hiking trails is one of the features that compels me to use other apps. (Well, maybe a separate layer of wikipedia tags as well.) We have quite a few marked trails in Poland, and it's a shame that this data is not used.

**Comment by gust001** [comment]
> I'm not sure it's any better here it's combining colors together 

I think, the best way still is mapy.cz approach https://mapy.cz/turisticka?x=13.5181282&y=49.2313148&z=18 
or relatively similar freemap.sk approach.

But anyway, showing hiking trails is one of the most important map applications together with car navigation.

**Comment by pastk** [comment]
> 2\. Comprehensive routes support - render as on-demand color-coded overlay lines, route over them, show distances, elevaton profiles, etc.
>         So routes should behave much like imported user tracks (and, possibly, guides?) and the implementation could be shared (e.g. it seems like Osmand uses exactly this approach).

We had a short talk about possible implementation and the tracks-like approach seems the way to go.
Then a lot of existing and future user tracks handling code could be re-used (displaying, analyzing (stats / elevation profiles..), following, editing, sharing etc.)

There are different options how to store routes data:
1. As kml/gpx files inside the map file (seems easier, but could inflate map sizes).
2. Add a relations-like support into map files, so that existing roads/paths/etc geometry is reused (more efficient, but more work also).
Also routes data could be optional to download (separate files) similar to 
- https://github.com/organicmaps/organicmaps/issues/8672


CC @rtsisyk @vng

**Comment by ikanakova** [comment]
> There are different options how to store routes data:
> 
> As kml/gpx files inside the map file (seems easier, but could inflate map sizes).
> Add a relations-like support into map files, so that existing roads/paths/etc geometry is reused (more efficient, but more work also).

I think option 2 would be better. I chose Organic maps, even though it doesn't have color-coded hiking routes, because it renders maps better than e.g. Osmand. So I'm worried about losing that feature.

**Comment by vng** [comment]
https://github.com/organicmaps/organicmaps/releases/tag/2025.08.06-4-android

So far, so good! Without shields, but with the primary color from symbol/osmc:symbol tags.
Feedback is welcome!

**Comment by Rob-from-VI** [comment]
It seems the opposite has been achieved in the latest google beta.

When I zoom my screen to just over 2.2  km wide using beta version, all trails disappear.

When I do the same in last released version, I can see many trails rendered all the way up to about 16km screen width.   This may be a bit too busy, but it is nice to be able to spot areas with many trails.

This above was done using Vancouver Island, BC Canada map download latest available for each version.

**Comment by vng** [comment]
- What is the beta app version?
- Do you have "Outdoor" style enabled in Release version?
- Please, give some screenshots.

**Comment by Rob-from-VI** [comment]
I realize I did not have the outdoor layer turned onin the beta version. All is well now.
